content/providers/provider-terraform/migrate-provider-tf.md

```diff
@@ -0,0 +1,256 @@
+in the file.
+
+The `vars` field allows you to create configuration variables to pass to your
+Terraform configuration.
+
+Create a new file called `terraform-configuration.yaml`. Your complete configuration
+file should match the following:
+
+```yaml
+apiVersion: tf.upbound.io/v1beta1
+kind: Workspace
+metadata:
+  name: tf-vm
+spec:
+  forProvider:
+	source: Inline
+	module: |
+
+	resource "aws_instance" "my_vm" {
+    	ami                   	= "ami-065deacbcaac64cf2"
+    	instance_type         	= "t2.micro"
+    	tags = {
+        	Name = var.vmName,
+    	}
+  	}
+	variable "vmName" {
+    	description = "VM name"
+    	type    	= string
+  	}
+
+	vars:
+  	- key: vmName
+    	value: crossplanevm
+```
+
+## Install the provider
+
+You wrote a managed resource in the previous step. Next, install the
+`provider-terraform` into your Kubernetes cluster with a Kubernetes
+configuration file.
+
+Create a new file called `provider-terraform-install.yaml`.
+
+Copy and paste the configuration below.
+
+```yaml
+apiVersion: pkg.crossplane.io/v1
+kind: Provider
+metadata:
+  name: provider-terraform
+spec:
+  package: xpkg.upbound.io/upbound/provider-terraform
+```
+
+Crossplane uses this Kubernetes manifest file to download and install the
+provider package into your cluster.
+
+Deploy the configuration file with `kubectl apply -f`
+
+`$ kubectl apply -f provider-terraform-install.yaml`
+
+Verify the provider with `kubectl get pods`.
+
+```yaml
+$ kubectl get pods -n upbound-system
+NAME                                                              READY   STATUS    RESTARTS   AGE
+crossplane-6979f579f9-x7nkr                                       2/2     Running   0          31m
+upbound-provider-aws-ec2-64262f355830-7b6b976c66-4b5wt            1/1     Running   0          29m
+upbound-provider-family-aws-fec919bd2218-5f85944578-rdxjm         1/1     Running   0          29m
+```
+
+
+## Authenticate with your cloud provider
+
+The Crossplane AWS provider configuration handles authentication. You must
+create a Kubernetes secret file to authenticate with your AWS account.
+
+Crossplane supports AWS authentication with:
+Authentication Keys
+Web Identity
+Service Accounts
+
+For more information on cloud provider authentication, checkout the Provider
+Azure and Provider GCP authentication documentation.
+
+In this example, the credentials.source attribute references the authentication
+key method as a secret.
+
+## Deploy your configuration
+
+Next, apply the manifest.
+
+`$ kubectl apply -f terraform-config.yaml`
+
+## State management
+
+For greenfield deployments, this approach works well for users who are already
+familiar with Terraform. If your organization wants to maintain state parity
+with a remote Terraform state file, you can use a `ProviderConfig` to point your
+Crossplane-controlled Terraform provider to the correct backend.
+
+The example below uses the `kubernetes` backend:
```

Comment:
`kubernetes` or other backend configuration is required before applying the Workspace MR, otherwise it fails. So maybe this part should be configured before applying Workspace in the previous step.